File: Data Structures & Algorithms/generate-parentheses/submission-1.java
class Solution {
    public List<String> generateParenthesis(int n) {
        List<String> res = new ArrayList<>();
        if (n == 0){
            return res;
        }
        Set<String> set = new HashSet<>();
        dfs("", n, n, set);
        for(var str : set){
            res.add(str);
        }
        return res;
    }

    private void dfs( String str, int left, int right, Set<String> set){
        if(left == 0 && right == 0){
            set.add(str);
            return;
        }

        if(left > 0){
            dfs(str + "(", left -1 ,right, set);
        }
        
        if (right > left){
            dfs(str + ")", left , right -1 , set);
        }
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = list()
        q = deque()
        q.append(root)
        if not root:
            return res
            
        while q:
            size = len(q)
            li = []
            for i in range(size):
                cur = q.popleft()
                li.append(cur.val)
                if cur.left:
                    q.append(cur.left)
                
                if cur.right:
                    q.append(cur.right)

            res.append(li)

        return res
